akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -109,6 +110,10 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
 
               dispatchSubstream(sub)
 
+            case RstStreamFrame(streamId, errorCode) ⇒
+              // FIXME: also need to handle the other case when no response has been produced yet (inlet still None)
+              incomingStreams(streamId).inlet.foreach(_.cancel())
```

Comment:
to confirm my understanding, the cancel only is for a given stream and the connection stays alive... Did not yet see a test for that, reviewing further, would be good to add if not covered yet

> The RST_STREAM frame fully terminates the referenced stream and causes it to enter the "closed" state. After receiving a RST_STREAM on a stream, the receiver MUST NOT send additional frames for that stream, with the exception of PRIORITY. However, after sending the RST_STREAM, the sending endpoint MUST be prepared to receive and process additional frames sent on the stream that might have been sent by the peer prior to the arrival of the RST_STREAM.